### PR TITLE
Introduce opaque_types for the WireCompiler

### PR DESCRIPTION
**Pull request by oldergod**

### CODE REVIEW

File: gen-tests.gradle.kts [hunk 1]
```diff
@@ -213,6 +213,26 @@ val generateAndroidCompactTests by tasks.creating(JavaExec::class) {
   )
 }
 
+// OPAQUE TYPES
+
+val generateOpaqueTypeTests by tasks.creating(JavaExec::class) {
+  group = "Generate Tests"
+  description = "Generates compact Java classes for Android from the test protos"
```
Comment (dnkoutso) [inline gen-tests.gradle.kts:220]: nit: is this description copy pasta'd?

File: wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt [hunk 2]
```diff
@@ -98,6 +98,11 @@ import okio.openZip
  *
  * The `--android-annotations` flag will add the `Nullable` annotation to optional fields.
  *
+ * The `opaque_types` takes a list of qualified named Protobuf types separated by a ','. All types
```
Comment (dnkoutso) [inline wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt:101]: nit maybe add `\`--opaque_types\`` to match the rest?

**Review by dnkoutso — APPROVED** [review]

File: gen-tests.gradle.kts (hunk 1 above)
Comment (oldergod) [reply gen-tests.gradle.kts]: fixed!

File: wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt (hunk 2 above)
Comment (oldergod) [reply wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt]: fixed!

**Pull request merged**